lib/partition_setup.pm

```diff
@@ -218,5 +218,81 @@ sub enable_encryption_guided_setup {
     installation_user_settings::await_password_check;
 }
 
+sub take_first_disk_storage_ng {
+    my (%args) = @_;
+    return unless is_storage_ng;
+    send_key $cmd{guidedsetup};    # select guided setup
+    assert_screen 'select-hard-disks';
+    # It's not always the case that SUT has 2 drives, for ipmi it's changing
+    # So making it flexible, still assert the screen if want to verify explicitly
+    assert_screen [qw(existing-partitions hard-disk-dev-sdb-selected)];
+    if (match_has_tag 'hard-disk-dev-sdb-selected' || get_var('SELECT_FIRST_DISK')) {
+        if (check_var('VIDEOMODE', 'text')) {
+            if (match_has_tag 'hotkey_d') {
+                send_key 'alt-d';
+            }
+            elsif (match_has_tag 'hotkey_e') {
+                send_key 'alt-e';
+            }
+            else {
+                die 'Needle needs tag \'hotkey_d\' or \'hotkey_e\'';
+            }
+        }
+        else {
+            assert_and_click 'hard-disk-dev-sdb-selected';    # Unselect second drive
+        }
+        assert_screen 'select-hard-disks-one-selected';
+        send_key $cmd{next};
+    }
+
+    # If drive is not formatted, we have select hard disks page
+    # On ipmi we always have unformatted drive
+    if (get_var('ISO_IN_EXTERNAL_DRIVE') || check_var('BACKEND', 'ipmi')) {
+        assert_screen 'existing-partitions';
+        send_key $cmd{next};
+    }
+    assert_screen 'partition-scheme';
+    send_key $cmd{next};
+    # select btrfs file system
+    if (check_var('VIDEOMODE', 'text')) {
+        assert_screen 'select-root-filesystem';
+        send_key 'alt-f';
+        send_key_until_needlematch 'filesystem-btrfs', 'down', 10, 3;
+        send_key 'ret';
+    }
+    else {
+        assert_and_click 'default-root-filesystem';
+        assert_and_click "filesystem-btrfs";
+    }
+    assert_screen "btrfs-selected";
+    send_key $cmd{next};
+}
+
+sub take_first_disk {
+    my (%args) = @_;
+    if (is_storage_ng) {
+        take_first_disk_storage_ng %args;
+        return;
```

Comment:
Why not use here an `else`?
`return` in the middle of a function is the new `goto` ;)

I could agree with the use of a return when there is a significant performance impact. But here it isn't.

Reply:
I agree, that it will be more clear with else here, so fixed. Thanks!